### ISSUE TITLE
Maximum Likelihood Estimation (MLE), Cross Entropy Loss (CE), and Negative Log-Likelihood (NLL)

### ISSUE THREAD

**Issue by HaFred** [HaFred/dnn-rocket-accelerator#47]
Refer to **Ian Goodfellow DL book** or CMU slides in the `book` folder for MLE.
Nice blog: https://leimao.github.io/blog/Cross-Entropy-KL-Divergence-MLE/ for the same purpose.

**Comment by HaFred** [comment]
* [Understanding softmax and the negative log-likelihood](https://ljvmiranda921.github.io/notebook/2017/08/13/softmax-and-the-negative-log-likelihood/) This blog talks about softmax and NLL
* [Understand Cross Entropy Loss in Minutes](https://medium.com/data-science-bootcamp/understand-cross-entropy-loss-in-minutes-9fb263caee9a)
* [比較 Cross Entropy 與 Mean Squared Error](https://medium.com/jarvis-toward-intelligence/%E6%AF%94%E8%BC%83-cross-entropy-%E8%88%87-mean-squared-error-8bebc0255f5)

**Comment by HaFred** [comment]
MLE is frequentist, whereas MAP is bayesian:
Wiki: _It is closely related to the method of maximum likelihood (ML) estimation, but employs an augmented optimization objective which incorporates a prior distribution (that quantifies the additional information available through prior knowledge of a related event) over the quantity one wants to estimate_

**Comment by HaFred** [comment]
## MLE & CE
MLE only deals with argmax`log p_model`. With label y bringing in, it is equivalent to argmin `-y_i logp_model`, which is the CE.

**Comment by HaFred** [comment]
# Bayesian
An interesting blog about the comparison:
https://towardsdatascience.com/frequentist-vs-bayesian-approaches-in-machine-learning-86ece21e820e

A pytorch-Bayesian-NN work https://github.com/kumar-shridhar/PyTorch-BayesianCNN

A hands-on survey paper for bayesian CNN https://arxiv.org/pdf/2007.06823.pdf
